android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -645,6 +774,101 @@ void createTouchScalar(ImageResolution resolution, DisplayMetrics displayMetrics
+                parameters.getResolution().getResolutionHeight())
+        );
+
+        // get the first one - the Desired resolution to guarantee streaming will start
+        validCapabilities.add(preferredCapability);
+        for (VideoStreamingRange range: ranges) {
+
+            Integer constraintHeightMax = range.getMaxResolution().getResolutionHeight();
+            Integer constraintHeightMin = range.getMinResolution().getResolutionHeight();
+
+            for (VideoStreamingCapability capability : originalAdditionalCapabilities) {
+                double diagonal;
+                if (capability.getPreferredResolution() == null
+                        || capability.getPreferredResolution().getResolutionHeight() == null
+                        || capability.getPreferredResolution().getResolutionWidth() == null) {
+                    continue;
+                }
+                if (capability.getDiagonalScreenSize() == null) {
+                    diagonal = parameters.getPreferredDiagonal();
+                } else {
+                    diagonal = capability.getDiagonalScreenSize();
+                }
+
+                if (range.getMinScreenDiagonal() > diagonal) {
+                    continue;
+                }
+
+                if (!isAspectRatioInRange(range.getAspectRatio(), capability.getPreferredResolution())) {
+                    if (constraintHeightMax == null && constraintHeightMin == null) {
+                        continue;
+                    }
+                }
+
+                if (!isImageResolutionInRange(range.getMinResolution(), range.getMaxResolution(), capability.getPreferredResolution())) {
+                    continue;
+                }
+
+                validCapabilities.add(capability);
+            }
+        }
+
+        return validCapabilities;
+    }
+
+    public Boolean isImageResolutionInRange(Resolution minResolution, Resolution maxResolution, ImageResolution currentResolution) {
+
+        Integer constraintHeightMax = maxResolution.getResolutionHeight();
+        Integer constraintHeightMin = minResolution.getResolutionHeight();
+        Integer constraintWidthMax = maxResolution.getResolutionWidth();
+        Integer constraintWidthMin = minResolution.getResolutionWidth();
+        Integer resolutionHeight = currentResolution.getResolutionHeight();
+        Integer resolutionWidth = currentResolution.getResolutionWidth();
+        if (currentResolution.getResolutionHeight() > 0 && currentResolution.getResolutionWidth() > 0 && constraintHeightMax != null && constraintHeightMin != null) {
+            if (!(resolutionHeight >= constraintHeightMin && resolutionHeight <= constraintHeightMax)) {
+                return false;
+            }
+
+            if (!(resolutionWidth >= constraintWidthMin && resolutionWidth <= constraintWidthMax)) {
+                return false;
+            }
+        }
+
+        // TODO check what if dev provided invalid constraints
```

Comment:
This TODO should be resolved before completion of this card, based on the iOS implementation we should return false if the range and constraints are invalid
```
if (currentResolution.getResolutionHeight() > 0 && currentResolution.getResolutionWidth() > 0 && constraintHeightMax != null && constraintHeightMin != null) {
            if (!(resolutionHeight >= constraintHeightMin && resolutionHeight <= constraintHeightMax)) {
                return false;
            }

            if (!(resolutionWidth >= constraintWidthMin && resolutionWidth <= constraintWidthMax)) {
                return false;
            }

            return true;
        } else {
            return false;
        }
```